### PR TITLE
[VL] feat: support partial merge phase in aggregation

### PR DESCRIPTION
**Pull request by rui-mo**
## What changes were proposed in this pull request?

Support partial merge phase on Velox backend. This PR mainly focuses on supporting the partial merge step brings by `count distinct`, and a follow-up PR will be proposed to support other distinct functions.
https://github.com/oap-project/velox/pull/193

## How was this patch tested?

Verified on Jenkins.

### CODE REVIEW

File: gluten-data/src/main/scala/io/glutenproject/execution/GlutenHashAggregateExecTransformer.scala [hunk 1]
```diff
@@ -181,28 +179,37 @@ case class GlutenHashAggregateExecTransformer(
...
+          .getTypeNode(veloxSixIntermediateTypes(5), nullable = false))
       case _: CovPopulation | _: CovSample =>
-        structTypeNodes.add(ConverterUtils.getTypeNode(DoubleType, nullable = false))
-        structTypeNodes.add(ConverterUtils.getTypeNode(LongType, nullable = false))
-        structTypeNodes.add(ConverterUtils.getTypeNode(DoubleType, nullable = false))
-        structTypeNodes.add(ConverterUtils.getTypeNode(DoubleType, nullable = false))
-        structTypeNodes.add(ConverterUtils.getTypeNode(DoubleType, nullable = true))
-        structTypeNodes.add(ConverterUtils.getTypeNode(LongType, nullable = true))
-        structTypeNodes.add(ConverterUtils.getTypeNode(DoubleType, nullable = true))
-        structTypeNodes.add(ConverterUtils.getTypeNode(DoubleType, nullable = true))
```
Comment (rui-mo) [inline gluten-data/src/main/scala/io/glutenproject/execution/GlutenHashAggregateExecTransformer.scala:202]: Hi @liujiayi771, could you help review this change? Eight intermediate types were used here, but looks to me only four is needed.

File: gluten-data/src/main/scala/io/glutenproject/execution/GlutenHashAggregateExecTransformer.scala [hunk 2]
```diff
@@ -127,9 +127,7 @@ case class GlutenHashAggregateExecTransformer(
           // Select sum from Velox Struct.
           expressionNodes.add(ExpressionBuilder.makeSelection(colIdx, 0))
           // Select isEmpty from Velox Struct.
-          expressionNodes.add(ExpressionBuilder
-            .makeCast(ConverterUtils.getTypeNode(BooleanType, nullable = true),
-              ExpressionBuilder.makeSelection(colIdx, 1), SQLConf.get.ansiEnabled))
```
Comment (rui-mo) [inline gluten-data/src/main/scala/io/glutenproject/execution/GlutenHashAggregateExecTransformer.scala:132]: Also use Boolean type back here. In my local testing, it works well. Do you remember the issue to solve here?

**Review by zhouyuan — APPROVED** [review]
👍  verified locally, Q95 can off-load hashagg operators with this patch

**Review by PHILO-HE — APPROVED** [review]

**Comment by rui-mo** [comment]
Thanks for your review!

File: gluten-data/src/main/scala/io/glutenproject/execution/GlutenHashAggregateExecTransformer.scala (hunk 1 above)
Comment (liujiayi771) [reply gluten-data/src/main/scala/io/glutenproject/execution/GlutenHashAggregateExecTransformer.scala:200]: Only need four, It should be that a rebase has retained the wrong code.

Comment (rui-mo) [reply gluten-data/src/main/scala/io/glutenproject/execution/GlutenHashAggregateExecTransformer.scala:200]: Got it. Thnaks for checking.

**Pull request merged**